tests/pact/fqns.repl

```diff
@@ -77,3 +77,165 @@
 
 (expect "selects correct test" (modB.get-test) "hello")
 (commit-tx)
+
+;;
+;; Module redeploy name resolution
+;;
+
+; In the following tests, we define a module `test-mod-redeploy-ref`, and then
+; redeploy the same module with the change to one capability: `test`.
+; In the old version, the `test` capability fails, in the new one it passes.
+
+(begin-tx)
+; First, demonstrate the behavior prior to pact-4.8.
+(env-exec-config ["DisablePact48"])
+
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+; Before pact-4.8, the updated capability will be ignored, and calls to a function
+; requiring that capability will fail. This is true whether we invoke `f` with or
+; without its module name, and with or without its namespace.
```

Comment:
I'm realizing a problem with this test overall which is the expectation triples (unqualified, un-namespaced, FQ) **never reveal any difference in behavior** and as such do not test anything. If you scan down, they always either **fail together** or **succeed together**. 

I suspect that there is some confusion here: the functionality under test is fixing whether module upgrades handle FQNs right **within module code**. The bug is never implicated by whether the calling context is qualified.

If I am correct, then pick one and remove the rest for all the tests.